### ISSUE TITLE
blockchain: accounting error (off-by-one?) in utxo set cache size 

### ISSUE THREAD

**Issue by Roasbeef** [btcsuite/btcd#2087]
Was running some sync tests with various configurations and spotted this upon restart: 
```
2024-01-01 16:48:11.662 [INF] CHAN: Flushing UTXO cache of 3127 MiB with 1 entries to disk. For large sizes, this can take up to several minutes...
```

1 entry, yet 3 GB cache size, smells like an accounting error....

**Comment by Roasbeef** [comment]
cc @kcalvinalvin

**Comment by kcalvinalvin** [comment]
Will take a look today but my initial assumption is that the 3GB part is how much the preallocated map is taking up.

The actual allocation happens like so:

1: User passes in the acceptable mem usage.
2: Pre-allocate a map based on this. It's usually like half of the mem usage (ex: 8gb of utxocache will alloc ~4gb initially)
3: utxo cache memory usage increases as each utxoentry is cached. The variable sized pkscript places a big part of this memory usage.

Did that happen at the very start?  If we're not in headers first, it'll try to "flush" the utxocache. If it was "flushing 1 entry" at the very first start that may be an off by one on the log message

**Comment by kcalvinalvin** [comment]
Fixed with #2089
